src/guarded.hpp

```diff
@@ -116,6 +116,8 @@ class guarded<T, M>::deleter
   public:
     using pointer = T *;
 
+    deleter();
```

Comment:
This has the side effect of allowing the user to default construct a guarded<T>::handle type. If this is a good idea, it is something which may need to be done in other classes as well.

Reply:
I have made another push, now the default deleter is removed